pilot/pkg/model/service.go

```diff
@@ -394,16 +404,16 @@ func (s *Service) External() bool {
 	return s.ExternalName != ""
 }
 
-// Key generates a unique string referencing service instances for a given port and labels.
+// DeprecatedKey generates a unique string referencing service instances for a given port and labels.
 // The separator character must be exclusive to the regular expressions allowed in the
 // service declaration.
-func (s *Service) Key(port *Port, labels Labels) string {
+func (s *Service) DeprecatedKey(port *Port, labels Labels) string {
```

Comment:
We probably shouldn't touch this (even for a rename), since it is used by the v1 v1alpha1 implementation.

Reply:
I updated all V1 code as well. The reason I renamed this was because I didn't want others to accidentally use this to construct the service key or cluster name. I can try to revert the naming